test/format/import.c

```diff
@@ -0,0 +1,212 @@
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "format.h"
+
+static void copy_file_into_directory(WT_SESSION *session, const char *dir, const char *name);
+static void import_with_repair(WT_SESSION *session);
+static void import_with_file_metadata(WT_SESSION *session, WT_SESSION *import_session);
+static void verify_import(WT_SESSION *session, int start_value);
+
+/*
+ * Import directory initialize command, remove and re-create the primary backup directory, plus a
+ * copy we maintain for recovery testing.
+ */
+#define HOME_IMPORT_INIT_CMD "rm -rf %s/" IMPORT_DIR "&& mkdir %s/" IMPORT_DIR
+#define IMPORT_DIR "IMPORT"
+#define IMPORT_URI "table:import"
+#define IMPORT_URI_CONFIG "key_format=i,value_format=i"
+#define IMPORT_ENTRIES 1000
+
+/*
+ * import --
+ *     Periodically import table.
+ */
+WT_THREAD_RET
+import(void *arg)
+{
+    WT_CONNECTION *conn, *import_conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION *import_session, *session;
+    size_t cmd_len;
+    uint32_t import_value;
+    u_int period;
+    int counter;
+    char cmd[64];
+
+    WT_UNUSED(arg);
+    conn = g.wts_conn;
+    counter = 0;
+    import_value = false;
+
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) * 2 + strlen(HOME_IMPORT_INIT_CMD) + 1;
+    testutil_check(__wt_snprintf(cmd, cmd_len, HOME_IMPORT_INIT_CMD, g.home, g.home));
+    testutil_checkfmt(system(cmd), "%s", "import directory creation failed");
+
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) + strlen(IMPORT_DIR) + 10;
+    testutil_check(__wt_snprintf(cmd, cmd_len, "%s/%s", g.home, IMPORT_DIR));
+    /* Open a connection to the database, creating it if necessary. */
+    create_database(cmd, &import_conn);
+
+    /* Open a session */
+    testutil_check(import_conn->open_session(import_conn, NULL, NULL, &import_session));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
```

Comment:
A bigger comment here to clarify and note that we're opening two sessions, one on the import database connection and one on the usual test database connection.

Reply:
Added comment.